src/main/java/de/tum/in/www1/artemis/web/rest/FileUploadSubmissionResource.java

```diff
@@ -33,81 +46,327 @@
+     * @param exerciseId the id of the exercise
+     * @param submittedOnly if only submitted submissions should be returned
+     * @param assessedByTutor if the submission was assessed by calling tutor
+     * @return the ResponseEntity with status 200 (OK) and the list of File Upload Submissions in body
      */
-    @GetMapping("/file-upload-submissions/{id}")
-    public ResponseEntity<FileUploadSubmission> getFileUploadSubmission(@PathVariable Long id) {
-        log.debug("REST request to get FileUploadSubmission : {}", id);
-        Optional<FileUploadSubmission> fileUploadSubmission = fileUploadSubmissionRepository.findById(id);
-        return ResponseUtil.wrapOrNotFound(fileUploadSubmission);
+    @GetMapping("/exercises/{exerciseId}/file-upload-submissions")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<List<FileUploadSubmission>> getAllFileUploadSubmissions(@PathVariable Long exerciseId, @RequestParam(defaultValue = "false") boolean submittedOnly,
+            @RequestParam(defaultValue = "false") boolean assessedByTutor) {
+        log.debug("REST request to get all file upload submissions");
+        Exercise exercise = exerciseService.findOne(exerciseId);
+
+        if (!authCheckService.isAtLeastTeachingAssistantForExercise(exercise)) {
+            throw new AccessForbiddenException("You are not allowed to access this resource");
+        }
+
+        List<FileUploadSubmission> fileUploadSubmissions;
+        if (assessedByTutor) {
+            User user = userService.getUserWithGroupsAndAuthorities();
+            fileUploadSubmissions = fileUploadSubmissionService.getAllFileUploadSubmissionsByTutorForExercise(exerciseId, user.getId());
+        }
+        else {
+            fileUploadSubmissions = fileUploadSubmissionService.getFileUploadSubmissions(exerciseId, submittedOnly);
+        }
+
+        return ResponseEntity.ok().body(fileUploadSubmissions);
+    }
+
+    /**
+     * GET /file-upload-submission-without-assessment : get one File Upload Submission without assessment.
+     *
+     * @param exerciseId the id of the exercise
+     * @return the ResponseEntity with status 200 (OK) and the list of File Upload Submissions in body
+     */
+    @GetMapping(value = "/exercises/{exerciseId}/file-upload-submission-without-assessment")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    @Transactional(readOnly = true)
+    public ResponseEntity<FileUploadSubmission> getFileUploadSubmissionWithoutAssessment(@PathVariable Long exerciseId) {
+        log.debug("REST request to get a file upload submission without assessment");
+        Exercise exercise = fileUploadExerciseService.findOne(exerciseId);
+
+        if (!authCheckService.isAtLeastTeachingAssistantForExercise(exercise)) {
+            return forbidden();
+        }
+        if (!(exercise instanceof FileUploadExercise)) {
```

Comment:
This condition is always false because fileUploadExerciseService.findOne(exerciseId) will retrieve a FileUploadExercise or null (but never another exercise type).

Reply:
Right. Deleted it